src/perl/Permabit/BlockDevice/VDO.pm

```diff
@@ -589,30 +592,37 @@ sub installModule {
       $self->uninstallModule($machineName);
     }
   }
-
+  # If useUpstreamModule is set, we need to use $VDO_LINUXNEXT_MODNAME(dm-vdo)
+  # module otherwise we load the $VDO_MODNAME(kvdo)
+  if ($self->{useUpstreamModule}) {
+    $moduleName = $VDO_LINUXNEXT_MODNAME;
+  }
   # Build the kernel and user tool modules from source.
-  $log->info("Installing module $moduleName $version on $machineName");
   my $module
     = Permabit::KernelModule->new(
                                   machine         => $self->getMachine(),
                                   modDir          => $self->getModuleSourceDir(),
                                   modFileName     => "kmod-$moduleName",
                                   modName         => $moduleName,
                                   modVersion      => $version,
+                                  upstreamModName => $VDO_LINUXNEXT_MODNAME,
```

Comment:
I don't understand what this upstreaModName argument is for. We already have modName...